### PR TITLE
[RPS-438] Setup files buckets for On Demand environments

### PR DESCRIPTION
**Pull request by wojtek-oledzki**
Each OD environment has it's own bucket, production S3 bucekt has a
CloudFront in front with digital.nhs.uk certificate.

Also includes all the manual changes (the horror!) I've applied in last
few weeks to the CD pipeline.

### CODE REVIEW

File: ansible/roles/foundations/aws_cd/files/code_pipeline_ps_hippo.yml [hunk 1]
```diff
@@ -35,6 +42,8 @@ Resources:
             Resource:
               - !Sub "arn:aws:s3:::artefacts.${HostedDomainPublic}"
```
Comment (wojtek-oledzki) [inline ansible/roles/foundations/aws_cd/files/code_pipeline_ps_hippo.yml]: remove - not used

File: ansible/roles/foundations/aws_ondemand/files/buckets.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+Parameters:
+  Environment:
+    Description: Environment Name
+    Type: String
+
+  FilesHost:
+    Type: String
+
+  HostedDomain:
```
Comment (wojtek-oledzki) [inline ansible/roles/foundations/aws_ondemand/files/buckets.yml]: not used - delete

**Review by wojtek-oledzki — COMMENTED** [review]
changes required

**Comment by wojtek-oledzki** [comment]
@ivorc / @ngta1 / @ziemsky - review comments applied and PR cleaned up a bit more.

**Review by ziemsky — APPROVED** [review]

**Pull request merged**